src/main/java/domain/pieces/Pieces.java

```diff
@@ -0,0 +1,83 @@
+package domain.pieces;
+
+import domain.pieces.exceptions.CanNotMoveException;
+import domain.point.Direction;
+import domain.point.Distance;
+import domain.point.Point;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+public class Pieces {
+	private Set<Piece> pieces;
+
+	public Pieces(Set<Piece> pieces) {
+		this.pieces = new HashSet<>(pieces);
+	}
+
+	public static Pieces of(PiecesFactory piecesFactory) {
+		return new Pieces(piecesFactory.getInstance());
+	}
+
+	public void move(Point from, Point to) {
+		Piece subject = find(from);
+		Direction direction = Direction.of(from, to);
+		Distance distance = Distance.of(from, to);
+		subject.canReach(distance);
+
+		if (direction.isLinearDirection()) {
+			subject = moveProgressivelyBeforeTo(from, to, subject, direction);
+		}
+
+		try {
+			Piece target = find(to);
+			attack(subject, target, direction);
+		} catch (CanNotMoveException e) {
+			moveOnly(subject, to, direction);
+		}
+	}
+
+	private Piece moveProgressivelyBeforeTo(Point from, Point to, Piece subject, Direction direction) {
+		Point point = from.add(direction.getRowIndex(), direction.getColumnIndex());
+		while (!point.equals(to)) {
+			subject = moveOnly(subject, point, direction);
+			point = point.add(direction.getRowIndex(), direction.getColumnIndex());
+		}
+		return subject;
+	}
+
+	private Piece find(Point point) {
+		return pieces.stream()
+					.filter(piece -> piece.getPoint().equals(point))
```

Comment:
```suggestion
					.filter(piece -> piece.matchPoint(point))
```

디미터의 법칙을 지킨다면 어떠한 점이 좋을지 고민해보세요.

Reply:
디미터의 법칙 또한, 소프트웨어의 결합도를 낮추기 위한 원칙 중 하나라고 생각합니다. 디미터의 법칙을 지키지 않으면 추후에 관련 로직을 수정할 때 이를 사용하는 많은 부분을 함께 수정해야하는 문제점이 생깁니다. 

디미터의 법칙을 알고는 있었으나 시간이 너무 촉박해서 지키지 못 했었습니다. ㅠㅠ